From 8daec2d9d3ff2d7ff85c203c0e9d07f5278b74f4 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Tue, 5 Sep 2017 15:20:44 -0500
Subject: [PATCH 015/105] net: Add SMC911X driver to Kconfig, convert

We add the various SMC91XX symbols to drivers/net/Kconfig and then this
converts the following to Kconfig:
   CONFIG_SMC911X
   CONFIG_SMC911X_BASE
   CONFIG_SMC911X_16_BIT
   CONFIG_SMC911X_32_BIT

Signed-off-by: Adam Ford <aford173@gmail.com>
[trini: Apply to the rest of the tree, re-squash old and new patch]
Signed-off-by: Tom Rini <trini@konsulko.com>
---
 README                                 | 15 ---------------
 board/compulab/cm_t35/cm_t35.c         |  3 ++-
 configs/ap325rxa_defconfig             |  4 ++++
 configs/arndale_defconfig              |  3 +++
 configs/blanche_defconfig              |  3 +++
 configs/cm_t3517_defconfig             |  4 ++++
 configs/cm_t35_defconfig               |  5 +++++
 configs/duovero_defconfig              |  4 ++++
 configs/igep0032_defconfig             |  4 ++++
 configs/igep00x0_defconfig             |  4 ++++
 configs/imx31_phycore_defconfig        |  4 ++++
 configs/imx31_phycore_eet_defconfig    |  4 ++++
 configs/kzm9g_defconfig                |  4 ++++
 configs/mx31pdk_defconfig              |  4 ++++
 configs/mx35pdk_defconfig              |  3 +++
 configs/mx53ard_defconfig              |  3 +++
 configs/odroid-xu3_defconfig           |  3 +++
 configs/omap3_evm_defconfig            |  4 ++++
 configs/omap3_logic_defconfig          |  4 ++++
 configs/omap3_overo_defconfig          |  5 +++++
 configs/omap3_zoom1_defconfig          |  4 ++++
 configs/peach-pi_defconfig             |  3 +++
 configs/peach-pit_defconfig            |  3 +++
 configs/smdk5250_defconfig             |  3 +++
 configs/smdk5420_defconfig             |  3 +++
 configs/smdkc100_defconfig             |  3 +++
 configs/smdkv310_defconfig             |  3 +++
 configs/snow_defconfig                 |  3 +++
 configs/spring_defconfig               |  3 +++
 configs/twister_defconfig              |  3 +++
 configs/uniphier_ld4_sld8_defconfig    |  3 +++
 configs/uniphier_v7_defconfig          |  4 ++++
 configs/uniphier_v8_defconfig          |  4 ++++
 configs/vct_platinum_defconfig         |  4 ++++
 configs/vct_platinum_onenand_defconfig |  4 ++++
 configs/vct_premium_defconfig          |  4 ++++
 configs/vct_premium_onenand_defconfig  |  4 ++++
 configs/vexpress_aemv8a_juno_defconfig |  4 ++++
 configs/vexpress_ca15_tc2_defconfig    |  4 ++++
 configs/vexpress_ca5x2_defconfig       |  4 ++++
 configs/vexpress_ca9x4_defconfig       |  4 ++++
 drivers/net/Kconfig                    | 28 ++++++++++++++++++++++++++++
 include/configs/ap325rxa.h             |  5 -----
 include/configs/blanche.h              |  4 ----
 include/configs/cm_t35.h               |  8 --------
 include/configs/cm_t3517.h             |  3 ---
 include/configs/duovero.h              |  4 ----
 include/configs/exynos5-common.h       |  3 ---
 include/configs/imx31_phycore.h        |  4 ----
 include/configs/kzm9g.h                |  4 ----
 include/configs/mx31pdk.h              |  4 ----
 include/configs/mx35pdk.h              |  4 ----
 include/configs/mx53ard.h              |  5 -----
 include/configs/omap3_evm.h            |  7 -------
 include/configs/omap3_igep00x0.h       |  9 ---------
 include/configs/omap3_logic.h          |  7 -------
 include/configs/omap3_overo.h          |  7 -------
 include/configs/omap3_zoom1.h          |  8 --------
 include/configs/r0p7734.h              |  5 -----
 include/configs/rsk7203.h              |  5 -----
 include/configs/rsk7264.h              |  5 -----
 include/configs/rsk7269.h              |  5 -----
 include/configs/smdkc100.h             |  3 ---
 include/configs/smdkv310.h             |  3 ---
 include/configs/twister.h              |  8 --------
 include/configs/uniphier.h             |  8 --------
 include/configs/vct.h                  | 22 ----------------------
 include/configs/vexpress_aemv8a.h      |  8 +-------
 include/configs/vexpress_common.h      |  6 ------
 scripts/config_whitelist.txt           |  5 -----
 70 files changed, 174 insertions(+), 184 deletions(-)

diff --git a/README b/README
index 242287ca9b..f288176228 100644
--- a/README
+++ b/README
@@ -1028,21 +1028,6 @@ The following options need to be configured:
 			control registers. This behavior won't affect the
 			correctnessof 10/100 link speed update.
 
-		CONFIG_SMC911X
-		Support for SMSC's LAN911x and LAN921x chips
-
-			CONFIG_SMC911X_BASE
-			Define this to hold the physical address
-			of the device (I/O space)
-
-			CONFIG_SMC911X_32_BIT
-			Define this if data bus is 32 bits
-
-			CONFIG_SMC911X_16_BIT
-			Define this if data bus is 16 bits. If your processor
-			automatically converts one 32 bit word to two 16 bit
-			words you may also try CONFIG_SMC911X_32_BIT.
-
 		CONFIG_SH_ETHER
 		Support for Renesas on-chip Ethernet controller
 
diff --git a/board/compulab/cm_t35/cm_t35.c b/board/compulab/cm_t35/cm_t35.c
index be938eb2cf..d5cfba4270 100644
--- a/board/compulab/cm_t35/cm_t35.c
+++ b/board/compulab/cm_t35/cm_t35.c
@@ -452,6 +452,7 @@ static int handle_mac_address(void)
  * Routine: board_eth_init
  * Description: initialize module and base-board Ethernet chips
  */
+#define SB_T35_SMC911X_BASE	(CONFIG_SMC911X_BASE + SZ_16M)
 int board_eth_init(bd_t *bis)
 {
 	int rc = 0, rc1 = 0;
@@ -460,7 +461,7 @@ int board_eth_init(bd_t *bis)
 	if (rc1)
 		printf("No MAC address found! ");
 
-	rc1 = cl_omap3_smc911x_init(0, 5, CM_T3X_SMC911X_BASE,
+	rc1 = cl_omap3_smc911x_init(0, 5, CONFIG_SMC911X_BASE,
 				    cm_t3x_reset_net_chip, -EINVAL);
 	if (rc1 > 0)
 		rc++;
diff --git a/configs/ap325rxa_defconfig b/configs/ap325rxa_defconfig
index 28443e2aa3..3f12093ef3 100644
--- a/configs/ap325rxa_defconfig
+++ b/configs/ap325rxa_defconfig
@@ -25,6 +25,10 @@ CONFIG_CMD_EXT2=y
 CONFIG_DOS_PARTITION=y
 CONFIG_ENV_IS_IN_FLASH=y
 CONFIG_MTD_NOR_FLASH=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0xB6080000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_BAUDRATE=38400
 CONFIG_SCIF_CONSOLE=y
 CONFIG_USE_PRIVATE_LIBGCC=y
diff --git a/configs/arndale_defconfig b/configs/arndale_defconfig
index f25fe7bf7d..708f06373e 100644
--- a/configs/arndale_defconfig
+++ b/configs/arndale_defconfig
@@ -25,6 +25,9 @@ CONFIG_DM_I2C_COMPAT=y
 CONFIG_MMC_DW=y
 CONFIG_MMC_SDHCI=y
 CONFIG_MMC_SDHCI_S5P=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x5000000
 CONFIG_SOUND=y
 CONFIG_I2S=y
 CONFIG_I2S_SAMSUNG=y
diff --git a/configs/blanche_defconfig b/configs/blanche_defconfig
index d0779285e5..500ebc6983 100644
--- a/configs/blanche_defconfig
+++ b/configs/blanche_defconfig
@@ -20,5 +20,8 @@ CONFIG_MTD_NOR_FLASH=y
 CONFIG_SPI_FLASH=y
 CONFIG_SPI_FLASH_BAR=y
 CONFIG_SPI_FLASH_SPANSION=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x18000000
 CONFIG_BAUDRATE=38400
 CONFIG_SCIF_CONSOLE=y
diff --git a/configs/cm_t3517_defconfig b/configs/cm_t3517_defconfig
index 8e7c07206b..dc0ded8269 100644
--- a/configs/cm_t3517_defconfig
+++ b/configs/cm_t3517_defconfig
@@ -38,6 +38,10 @@ CONFIG_LED_STATUS_BOOT_ENABLE=y
 CONFIG_LED_STATUS_BOOT=0
 CONFIG_MMC_OMAP_HS=y
 CONFIG_NAND=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x2D000000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_SYS_NS16550=y
 CONFIG_OMAP3_SPI=y
 CONFIG_USB=y
diff --git a/configs/cm_t35_defconfig b/configs/cm_t35_defconfig
index 1515c9253c..90950d4247 100644
--- a/configs/cm_t35_defconfig
+++ b/configs/cm_t35_defconfig
@@ -40,6 +40,10 @@ CONFIG_LED_STATUS_BOOT=0
 CONFIG_MMC_OMAP_HS=y
 CONFIG_NAND=y
 CONFIG_SPL_NAND_SIMPLE=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x2C000000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_SYS_NS16550=y
 CONFIG_OMAP3_SPI=y
 CONFIG_USB=y
@@ -47,3 +51,4 @@ CONFIG_USB_EHCI_HCD=y
 CONFIG_USB_STORAGE=y
 CONFIG_LCD=y
 CONFIG_OF_LIBFDT=y
+CONFIG_NETDEVICES=y
diff --git a/configs/duovero_defconfig b/configs/duovero_defconfig
index feec2cfb1a..b76da3e95a 100644
--- a/configs/duovero_defconfig
+++ b/configs/duovero_defconfig
@@ -23,6 +23,10 @@ CONFIG_CMD_EXT4_WRITE=y
 # CONFIG_SPL_EFI_PARTITION is not set
 CONFIG_SPL_PARTITION_UUIDS=y
 CONFIG_MMC_OMAP_HS=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x2C000000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_SYS_NS16550=y
 CONFIG_OMAP3_SPI=y
 CONFIG_USB=y
diff --git a/configs/igep0032_defconfig b/configs/igep0032_defconfig
index b99b6dbe42..0521039453 100644
--- a/configs/igep0032_defconfig
+++ b/configs/igep0032_defconfig
@@ -32,6 +32,10 @@ CONFIG_MMC_OMAP_HS=y
 CONFIG_NAND=y
 CONFIG_SYS_NAND_BUSWIDTH_16BIT=y
 CONFIG_SPL_NAND_SIMPLE=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x2C000000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_SYS_NS16550=y
 CONFIG_OMAP3_SPI=y
 CONFIG_FAT_WRITE=y
diff --git a/configs/igep00x0_defconfig b/configs/igep00x0_defconfig
index 0e2ab723c2..a2f29ccfd4 100644
--- a/configs/igep00x0_defconfig
+++ b/configs/igep00x0_defconfig
@@ -33,6 +33,10 @@ CONFIG_NAND=y
 CONFIG_SYS_NAND_BUSWIDTH_16BIT=y
 CONFIG_SPL_NAND_SIMPLE=y
 CONFIG_MTD_UBI_FASTMAP=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x2C000000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_SYS_NS16550=y
 CONFIG_OMAP3_SPI=y
 CONFIG_FAT_WRITE=y
diff --git a/configs/imx31_phycore_defconfig b/configs/imx31_phycore_defconfig
index 7b17b663a4..6be9913134 100644
--- a/configs/imx31_phycore_defconfig
+++ b/configs/imx31_phycore_defconfig
@@ -10,3 +10,7 @@ CONFIG_CMD_PING=y
 CONFIG_ENV_IS_IN_EEPROM=y
 # CONFIG_MMC is not set
 CONFIG_MTD_NOR_FLASH=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0xa8000000
+CONFIG_SMC911X_32_BIT=y
diff --git a/configs/imx31_phycore_eet_defconfig b/configs/imx31_phycore_eet_defconfig
index b67dcd3d0a..a8f57ef6cd 100644
--- a/configs/imx31_phycore_eet_defconfig
+++ b/configs/imx31_phycore_eet_defconfig
@@ -14,3 +14,7 @@ CONFIG_CMD_BMP=y
 CONFIG_ENV_IS_IN_EEPROM=y
 # CONFIG_MMC is not set
 CONFIG_MTD_NOR_FLASH=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0xa8000000
+CONFIG_SMC911X_32_BIT=y
diff --git a/configs/kzm9g_defconfig b/configs/kzm9g_defconfig
index f4b8ccdf2f..6271aa0925 100644
--- a/configs/kzm9g_defconfig
+++ b/configs/kzm9g_defconfig
@@ -17,6 +17,10 @@ CONFIG_CMD_FAT=y
 CONFIG_ENV_IS_IN_FLASH=y
 # CONFIG_MMC is not set
 CONFIG_MTD_NOR_FLASH=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x10000000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_SCIF_CONSOLE=y
 # CONFIG_FAT_WRITE is not set
 CONFIG_OF_LIBFDT=y
diff --git a/configs/mx31pdk_defconfig b/configs/mx31pdk_defconfig
index 3fcc0c0b5a..0884caf050 100644
--- a/configs/mx31pdk_defconfig
+++ b/configs/mx31pdk_defconfig
@@ -18,3 +18,7 @@ CONFIG_ENV_IS_IN_NAND=y
 # CONFIG_MMC is not set
 CONFIG_NAND=y
 CONFIG_NAND_MXC=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0xB6000000
+CONFIG_SMC911X_32_BIT=y
diff --git a/configs/mx35pdk_defconfig b/configs/mx35pdk_defconfig
index 2c3455f7e6..d4a567cc13 100644
--- a/configs/mx35pdk_defconfig
+++ b/configs/mx35pdk_defconfig
@@ -25,6 +25,9 @@ CONFIG_ENV_IS_IN_FLASH=y
 CONFIG_MTD_NOR_FLASH=y
 CONFIG_NAND=y
 CONFIG_NAND_MXC=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0xB6000000
 CONFIG_USB=y
 CONFIG_USB_EHCI_HCD=y
 CONFIG_USB_STORAGE=y
diff --git a/configs/mx53ard_defconfig b/configs/mx53ard_defconfig
index 1d8e5bc7fe..804e4cfedf 100644
--- a/configs/mx53ard_defconfig
+++ b/configs/mx53ard_defconfig
@@ -16,4 +16,7 @@ CONFIG_CMD_FAT=y
 CONFIG_ENV_IS_IN_MMC=y
 CONFIG_NAND=y
 CONFIG_NAND_MXC=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0xF4000000
 CONFIG_OF_LIBFDT=y
diff --git a/configs/odroid-xu3_defconfig b/configs/odroid-xu3_defconfig
index 5481188e64..76b1d35b2a 100644
--- a/configs/odroid-xu3_defconfig
+++ b/configs/odroid-xu3_defconfig
@@ -29,6 +29,9 @@ CONFIG_DFU_MMC=y
 CONFIG_MMC_DW=y
 CONFIG_MMC_SDHCI=y
 CONFIG_MMC_SDHCI_S5P=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x5000000
 CONFIG_DM_PMIC=y
 CONFIG_PMIC_S2MPS11=y
 CONFIG_DM_REGULATOR=y
diff --git a/configs/omap3_evm_defconfig b/configs/omap3_evm_defconfig
index 55b1e8d46a..5103e9fd8d 100644
--- a/configs/omap3_evm_defconfig
+++ b/configs/omap3_evm_defconfig
@@ -44,6 +44,10 @@ CONFIG_MTD=y
 CONFIG_NAND=y
 CONFIG_SYS_NAND_BUSWIDTH_16BIT=y
 CONFIG_SPL_NAND_SIMPLE=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x2C000000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_DM_SERIAL=y
 CONFIG_SYS_NS16550=y
 CONFIG_OMAP3_SPI=y
diff --git a/configs/omap3_logic_defconfig b/configs/omap3_logic_defconfig
index cdbebf8026..fc3c6fead9 100644
--- a/configs/omap3_logic_defconfig
+++ b/configs/omap3_logic_defconfig
@@ -37,6 +37,10 @@ CONFIG_SYS_NAND_BUSWIDTH_16BIT=y
 CONFIG_SYS_NAND_U_BOOT_LOCATIONS=y
 CONFIG_SYS_NAND_U_BOOT_OFFS=0x80000
 CONFIG_SPL_NAND_SIMPLE=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x08000000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_SYS_NS16550=y
 CONFIG_OMAP3_SPI=y
 CONFIG_USB=y
diff --git a/configs/omap3_overo_defconfig b/configs/omap3_overo_defconfig
index 237d5bdd11..389e72ab23 100644
--- a/configs/omap3_overo_defconfig
+++ b/configs/omap3_overo_defconfig
@@ -43,6 +43,10 @@ CONFIG_MMC_OMAP_HS=y
 CONFIG_NAND=y
 CONFIG_SYS_NAND_BUSWIDTH_16BIT=y
 CONFIG_SPL_NAND_SIMPLE=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x2C000000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_SYS_NS16550=y
 CONFIG_OMAP3_SPI=y
 CONFIG_USB=y
@@ -51,3 +55,4 @@ CONFIG_USB_STORAGE=y
 CONFIG_FAT_WRITE=y
 CONFIG_BCH=y
 CONFIG_OF_LIBFDT=y
+CONFIG_NETDEVICES=y
diff --git a/configs/omap3_zoom1_defconfig b/configs/omap3_zoom1_defconfig
index 3afd7e9722..bedbbd3221 100644
--- a/configs/omap3_zoom1_defconfig
+++ b/configs/omap3_zoom1_defconfig
@@ -33,6 +33,10 @@ CONFIG_ENV_IS_IN_NAND=y
 CONFIG_MMC_OMAP_HS=y
 CONFIG_NAND=y
 CONFIG_SYS_NAND_BUSWIDTH_16BIT=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x08000000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_SYS_NS16550=y
 CONFIG_OMAP3_SPI=y
 CONFIG_FAT_WRITE=y
diff --git a/configs/peach-pi_defconfig b/configs/peach-pi_defconfig
index 6900f3d77f..26b31099e7 100644
--- a/configs/peach-pi_defconfig
+++ b/configs/peach-pi_defconfig
@@ -38,6 +38,9 @@ CONFIG_MMC_SDHCI_S5P=y
 CONFIG_SPI_FLASH=y
 CONFIG_SPI_FLASH_GIGADEVICE=y
 CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x5000000
 CONFIG_DM_PMIC=y
 CONFIG_PMIC_TPS65090=y
 CONFIG_DM_REGULATOR=y
diff --git a/configs/peach-pit_defconfig b/configs/peach-pit_defconfig
index b073d820a3..daa263ed06 100644
--- a/configs/peach-pit_defconfig
+++ b/configs/peach-pit_defconfig
@@ -38,6 +38,9 @@ CONFIG_MMC_SDHCI_S5P=y
 CONFIG_SPI_FLASH=y
 CONFIG_SPI_FLASH_GIGADEVICE=y
 CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x5000000
 CONFIG_DM_PMIC=y
 CONFIG_PMIC_TPS65090=y
 CONFIG_DM_REGULATOR=y
diff --git a/configs/smdk5250_defconfig b/configs/smdk5250_defconfig
index 57870e5a9f..b6c8d4ce46 100644
--- a/configs/smdk5250_defconfig
+++ b/configs/smdk5250_defconfig
@@ -34,6 +34,9 @@ CONFIG_MMC_SDHCI_S5P=y
 CONFIG_SPI_FLASH=y
 CONFIG_SPI_FLASH_GIGADEVICE=y
 CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x5000000
 CONFIG_DM_PMIC=y
 CONFIG_DM_PMIC_MAX77686=y
 CONFIG_DM_REGULATOR=y
diff --git a/configs/smdk5420_defconfig b/configs/smdk5420_defconfig
index 75b8c00e28..5742d16823 100644
--- a/configs/smdk5420_defconfig
+++ b/configs/smdk5420_defconfig
@@ -29,6 +29,9 @@ CONFIG_MMC_SDHCI_S5P=y
 CONFIG_SPI_FLASH=y
 CONFIG_SPI_FLASH_GIGADEVICE=y
 CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x5000000
 CONFIG_DM_PMIC=y
 CONFIG_DM_REGULATOR=y
 CONFIG_EXYNOS_SPI=y
diff --git a/configs/smdkc100_defconfig b/configs/smdkc100_defconfig
index eef4aee919..221464074d 100644
--- a/configs/smdkc100_defconfig
+++ b/configs/smdkc100_defconfig
@@ -16,3 +16,6 @@ CONFIG_CMD_FAT=y
 CONFIG_CMD_MTDPARTS=y
 CONFIG_ENV_IS_IN_ONENAND=y
 # CONFIG_MMC is not set
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x98800300
diff --git a/configs/smdkv310_defconfig b/configs/smdkv310_defconfig
index 31a5e99a40..7cbdd4dd07 100644
--- a/configs/smdkv310_defconfig
+++ b/configs/smdkv310_defconfig
@@ -28,5 +28,8 @@ CONFIG_MMC_DW=y
 CONFIG_MMC_SDHCI=y
 CONFIG_MMC_SDHCI_SDMA=y
 CONFIG_MMC_SDHCI_S5P=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x5000000
 CONFIG_USB=y
 CONFIG_DM_USB=y
diff --git a/configs/snow_defconfig b/configs/snow_defconfig
index 66255ad492..72fd8c2192 100644
--- a/configs/snow_defconfig
+++ b/configs/snow_defconfig
@@ -42,6 +42,9 @@ CONFIG_MMC_SDHCI_S5P=y
 CONFIG_SPI_FLASH=y
 CONFIG_SPI_FLASH_GIGADEVICE=y
 CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x5000000
 CONFIG_DM_PMIC=y
 CONFIG_DM_PMIC_MAX77686=y
 CONFIG_PMIC_S5M8767=y
diff --git a/configs/spring_defconfig b/configs/spring_defconfig
index b5db3833cd..4eed491d1d 100644
--- a/configs/spring_defconfig
+++ b/configs/spring_defconfig
@@ -42,6 +42,9 @@ CONFIG_MMC_SDHCI_S5P=y
 CONFIG_SPI_FLASH=y
 CONFIG_SPI_FLASH_GIGADEVICE=y
 CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x5000000
 CONFIG_DM_PMIC=y
 CONFIG_DM_PMIC_MAX77686=y
 CONFIG_PMIC_S5M8767=y
diff --git a/configs/twister_defconfig b/configs/twister_defconfig
index ed463f5d32..90b2fde91f 100644
--- a/configs/twister_defconfig
+++ b/configs/twister_defconfig
@@ -34,6 +34,9 @@ CONFIG_NAND=y
 CONFIG_NAND_OMAP_GPMC_PREFETCH=y
 CONFIG_SYS_NAND_BUSWIDTH_16BIT=y
 CONFIG_SPL_NAND_SIMPLE=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x2C000000
 CONFIG_SYS_NS16550=y
 CONFIG_USB=y
 CONFIG_USB_EHCI_HCD=y
diff --git a/configs/uniphier_ld4_sld8_defconfig b/configs/uniphier_ld4_sld8_defconfig
index cd0efe371b..74c52ddd82 100644
--- a/configs/uniphier_ld4_sld8_defconfig
+++ b/configs/uniphier_ld4_sld8_defconfig
@@ -44,6 +44,9 @@ CONFIG_NAND_DENALI_DT=y
 CONFIG_SYS_NAND_DENALI_64BIT=y
 CONFIG_NAND_DENALI_SPARE_AREA_SKIP_BYTES=8
 CONFIG_SPL_NAND_DENALI=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_32_BIT=y
 CONFIG_USB=y
 CONFIG_USB_EHCI_HCD=y
 CONFIG_USB_EHCI_GENERIC=y
diff --git a/configs/uniphier_v7_defconfig b/configs/uniphier_v7_defconfig
index 7f2347a2f5..5c1d343691 100644
--- a/configs/uniphier_v7_defconfig
+++ b/configs/uniphier_v7_defconfig
@@ -43,6 +43,10 @@ CONFIG_NAND_DENALI_DT=y
 CONFIG_SYS_NAND_DENALI_64BIT=y
 CONFIG_NAND_DENALI_SPARE_AREA_SKIP_BYTES=8
 CONFIG_SPL_NAND_DENALI=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x0
+CONFIG_SMC911X_32_BIT=y
 CONFIG_USB=y
 CONFIG_USB_XHCI_HCD=y
 CONFIG_USB_EHCI_HCD=y
diff --git a/configs/uniphier_v8_defconfig b/configs/uniphier_v8_defconfig
index 31dd9b31cf..bc4bbbfbdb 100644
--- a/configs/uniphier_v8_defconfig
+++ b/configs/uniphier_v8_defconfig
@@ -38,6 +38,10 @@ CONFIG_NAND_DENALI=y
 CONFIG_NAND_DENALI_DT=y
 CONFIG_SYS_NAND_DENALI_64BIT=y
 CONFIG_NAND_DENALI_SPARE_AREA_SKIP_BYTES=8
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x0
+CONFIG_SMC911X_32_BIT=y
 CONFIG_SYSRESET=y
 CONFIG_SYSRESET_PSCI=y
 CONFIG_USB=y
diff --git a/configs/vct_platinum_defconfig b/configs/vct_platinum_defconfig
index 38b3e81ebc..0abf81b96a 100644
--- a/configs/vct_platinum_defconfig
+++ b/configs/vct_platinum_defconfig
@@ -15,6 +15,10 @@ CONFIG_CMD_SNTP=y
 CONFIG_CMD_FAT=y
 CONFIG_ENV_IS_IN_FLASH=y
 CONFIG_MTD_NOR_FLASH=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x00000000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_SYS_NS16550=y
 CONFIG_USB=y
 CONFIG_USB_EHCI_HCD=y
diff --git a/configs/vct_platinum_onenand_defconfig b/configs/vct_platinum_onenand_defconfig
index 6b57ac51cc..9c41e3943c 100644
--- a/configs/vct_platinum_onenand_defconfig
+++ b/configs/vct_platinum_onenand_defconfig
@@ -19,6 +19,10 @@ CONFIG_CMD_JFFS2=y
 CONFIG_CMD_UBI=y
 # CONFIG_CMD_UBIFS is not set
 CONFIG_ENV_IS_IN_ONENAND=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x00000000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_SYS_NS16550=y
 CONFIG_USB=y
 CONFIG_USB_EHCI_HCD=y
diff --git a/configs/vct_premium_defconfig b/configs/vct_premium_defconfig
index d962fd1206..c6219fa78b 100644
--- a/configs/vct_premium_defconfig
+++ b/configs/vct_premium_defconfig
@@ -15,6 +15,10 @@ CONFIG_CMD_SNTP=y
 CONFIG_CMD_FAT=y
 CONFIG_ENV_IS_IN_FLASH=y
 CONFIG_MTD_NOR_FLASH=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x00000000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_SYS_NS16550=y
 CONFIG_USB=y
 CONFIG_USB_EHCI_HCD=y
diff --git a/configs/vct_premium_onenand_defconfig b/configs/vct_premium_onenand_defconfig
index a55cb5b385..5cfbebf677 100644
--- a/configs/vct_premium_onenand_defconfig
+++ b/configs/vct_premium_onenand_defconfig
@@ -19,6 +19,10 @@ CONFIG_CMD_JFFS2=y
 CONFIG_CMD_UBI=y
 # CONFIG_CMD_UBIFS is not set
 CONFIG_ENV_IS_IN_ONENAND=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x00000000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_SYS_NS16550=y
 CONFIG_USB=y
 CONFIG_USB_EHCI_HCD=y
diff --git a/configs/vexpress_aemv8a_juno_defconfig b/configs/vexpress_aemv8a_juno_defconfig
index 9c3ee73c0a..b48e749e28 100644
--- a/configs/vexpress_aemv8a_juno_defconfig
+++ b/configs/vexpress_aemv8a_juno_defconfig
@@ -28,5 +28,9 @@ CONFIG_ENV_IS_IN_FLASH=y
 CONFIG_DM=y
 # CONFIG_MMC is not set
 CONFIG_MTD_NOR_FLASH=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x018000000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_DM_SERIAL=y
 CONFIG_OF_LIBFDT=y
diff --git a/configs/vexpress_ca15_tc2_defconfig b/configs/vexpress_ca15_tc2_defconfig
index a0032ade55..9303f89bb4 100644
--- a/configs/vexpress_ca15_tc2_defconfig
+++ b/configs/vexpress_ca15_tc2_defconfig
@@ -19,5 +19,9 @@ CONFIG_CMD_MMC=y
 # CONFIG_CMD_MISC is not set
 CONFIG_ENV_IS_IN_FLASH=y
 CONFIG_MTD_NOR_FLASH=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x1a000000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_BAUDRATE=38400
 CONFIG_OF_LIBFDT=y
diff --git a/configs/vexpress_ca5x2_defconfig b/configs/vexpress_ca5x2_defconfig
index efa8c31ea6..3f8bc9cc13 100644
--- a/configs/vexpress_ca5x2_defconfig
+++ b/configs/vexpress_ca5x2_defconfig
@@ -18,5 +18,9 @@ CONFIG_CMD_MMC=y
 # CONFIG_CMD_MISC is not set
 CONFIG_ENV_IS_IN_FLASH=y
 CONFIG_MTD_NOR_FLASH=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x1a000000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_BAUDRATE=38400
 CONFIG_OF_LIBFDT=y
diff --git a/configs/vexpress_ca9x4_defconfig b/configs/vexpress_ca9x4_defconfig
index 4e7b961997..4cbc962373 100644
--- a/configs/vexpress_ca9x4_defconfig
+++ b/configs/vexpress_ca9x4_defconfig
@@ -18,5 +18,9 @@ CONFIG_CMD_MMC=y
 # CONFIG_CMD_MISC is not set
 CONFIG_ENV_IS_IN_FLASH=y
 CONFIG_MTD_NOR_FLASH=y
+CONFIG_NETDEVICES=y
+CONFIG_SMC911X=y
+CONFIG_SMC911X_BASE=0x4e000000
+CONFIG_SMC911X_32_BIT=y
 CONFIG_BAUDRATE=38400
 CONFIG_OF_LIBFDT=y
diff --git a/drivers/net/Kconfig b/drivers/net/Kconfig
index d67927cd3b..c1ce54ecbb 100644
--- a/drivers/net/Kconfig
+++ b/drivers/net/Kconfig
@@ -211,6 +211,34 @@ config RTL8169
 	  This driver supports Realtek 8169 series gigabit ethernet family of
 	  PCI/PCIe chipsets/adapters.
 
+config SMC911X
+	bool "SMSC LAN911x and LAN921x controller driver"
+
+if SMC911X
+
+config SMC911X_BASE
+	hex "SMC911X Base Address"
+	help
+	  Define this to hold the physical address
+	  of the device (I/O space)
+
+choice
+	prompt "SMC911X bus width"
+	default SMC911X_16_BIT
+
+config SMC911X_32_BIT
+	bool "Enable 32-bit interface"
+
+config SMC911X_16_BIT
+	bool "Enable 16-bit interface"
+	help
+	  Define this if data bus is 16 bits. If your processor
+	  automatically converts one 32 bit word to two 16 bit
+	  words you may also try CONFIG_SMC911X_32_BIT.
+
+endchoice
+endif #SMC911X
+
 config SUN7I_GMAC
 	bool "Enable Allwinner GMAC Ethernet support"
 	help
diff --git a/include/configs/ap325rxa.h b/include/configs/ap325rxa.h
index 1140a70b5e..c09769dbed 100644
--- a/include/configs/ap325rxa.h
+++ b/include/configs/ap325rxa.h
@@ -16,11 +16,6 @@
 #define CONFIG_DISPLAY_BOARDINFO
 #undef  CONFIG_SHOW_BOOT_PROGRESS
 
-/* SMC9118 */
-#define CONFIG_SMC911X 1
-#define CONFIG_SMC911X_32_BIT 1
-#define CONFIG_SMC911X_BASE 0xB6080000
-
 /* MEMORY */
 #define AP325RXA_SDRAM_BASE		(0x88000000)
 #define AP325RXA_FLASH_BASE_1		(0xA0000000)
diff --git a/include/configs/blanche.h b/include/configs/blanche.h
index c09d430d3a..2b18033e64 100755
--- a/include/configs/blanche.h
+++ b/include/configs/blanche.h
@@ -65,11 +65,7 @@
 #undef  CONFIG_CMD_SPI
 #endif
 
-/* BLANCHE on board LANC: SMC89218 (ExCS0) */
 #define CONFIG_NET_MULTI
-#define CONFIG_SMC911X                  1
-#define CONFIG_SMC911X_16_BIT           1
-#define CONFIG_SMC911X_BASE             0x18000000
 
 /* Board Clock */
 #define RMOBILE_XTAL_CLK	20000000u
diff --git a/include/configs/cm_t35.h b/include/configs/cm_t35.h
index d0b990cb04..7cb3124153 100644
--- a/include/configs/cm_t35.h
+++ b/include/configs/cm_t35.h
@@ -207,14 +207,6 @@
 #define CONFIG_ENV_OFFSET		0x260000
 #define CONFIG_ENV_ADDR			0x260000
 
-#if defined(CONFIG_CMD_NET)
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_32_BIT
-#define CM_T3X_SMC911X_BASE	0x2C000000
-#define SB_T35_SMC911X_BASE	(CM_T3X_SMC911X_BASE + (16 << 20))
-#define CONFIG_SMC911X_BASE	CM_T3X_SMC911X_BASE
-#endif /* (CONFIG_CMD_NET) */
-
 /* additions for new relocation code, must be added to all boards */
 #define CONFIG_SYS_SDRAM_BASE		PHYS_SDRAM_1
 #define CONFIG_SYS_INIT_RAM_ADDR	0x4020f800
diff --git a/include/configs/cm_t3517.h b/include/configs/cm_t3517.h
index 4a31944e4b..fbb1135697 100644
--- a/include/configs/cm_t3517.h
+++ b/include/configs/cm_t3517.h
@@ -213,9 +213,6 @@
 #define CONFIG_DRIVER_TI_EMAC
 #define CONFIG_DRIVER_TI_EMAC_USE_RMII
 #define CONFIG_MII
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_32_BIT
-#define CONFIG_SMC911X_BASE	(0x2C000000 + (16 << 20))
 #define CONFIG_ARP_TIMEOUT		200UL
 #define CONFIG_NET_RETRY_COUNT		5
 #endif /* CONFIG_CMD_NET */
diff --git a/include/configs/duovero.h b/include/configs/duovero.h
index c4496a7f48..96644b1fa6 100644
--- a/include/configs/duovero.h
+++ b/include/configs/duovero.h
@@ -30,10 +30,6 @@
 
 #define CONFIG_SYS_ENABLE_PADS_ALL
 
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_32_BIT
-#define CONFIG_SMC911X_BASE		0x2C000000
-
 /* GPIO */
 
 /* ENV related config options */
diff --git a/include/configs/exynos5-common.h b/include/configs/exynos5-common.h
index 2860d6bb9b..57101b671d 100644
--- a/include/configs/exynos5-common.h
+++ b/include/configs/exynos5-common.h
@@ -123,9 +123,6 @@
 
 /* Ethernet Controllor Driver */
 #ifdef CONFIG_CMD_NET
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_BASE		0x5000000
-#define CONFIG_SMC911X_16_BIT
 #define CONFIG_ENV_SROM_BANK		1
 #endif /*CONFIG_CMD_NET*/
 
diff --git a/include/configs/imx31_phycore.h b/include/configs/imx31_phycore.h
index 1750b228a1..36fda47813 100644
--- a/include/configs/imx31_phycore.h
+++ b/include/configs/imx31_phycore.h
@@ -85,10 +85,6 @@
 		"pclk:185925,le:9,ri:17,up:7,lo:10,hs:1,vs:1,"		\
 		"sync:1241513985,vmode:0\0"
 
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_BASE	0xa8000000
-#define CONFIG_SMC911X_32_BIT
-
 /*
  * Miscellaneous configurable options
  */
diff --git a/include/configs/kzm9g.h b/include/configs/kzm9g.h
index a260d0e8ac..16f37a016d 100644
--- a/include/configs/kzm9g.h
+++ b/include/configs/kzm9g.h
@@ -103,10 +103,6 @@
 #define CONFIG_SH_SCIF_CLK_FREQ CONFIG_SYS_CLK_FREQ
 #define TMU_CLK_DIVIDER		(4)	/* 4 (default), 16, 64, 256 or 1024 */
 
-/* Ether */
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_BASE	(0x10000000)
-#define CONFIG_SMC911X_32_BIT
 #define CONFIG_NFS_TIMEOUT 10000UL
 
 /* I2C */
diff --git a/include/configs/mx31pdk.h b/include/configs/mx31pdk.h
index a9bfec2c49..3259e82761 100644
--- a/include/configs/mx31pdk.h
+++ b/include/configs/mx31pdk.h
@@ -79,10 +79,6 @@
 		"nand erase 0x0 0x40000; "				\
 		"nand write 0x81000000 0x0 0x40000\0"
 
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_BASE	0xB6000000
-#define CONFIG_SMC911X_32_BIT
-
 /*
  * Miscellaneous configurable options
  */
diff --git a/include/configs/mx35pdk.h b/include/configs/mx35pdk.h
index 18695d2686..8cc5163258 100644
--- a/include/configs/mx35pdk.h
+++ b/include/configs/mx35pdk.h
@@ -85,10 +85,6 @@
 /*
  * Ethernet on the debug board (SMC911)
  */
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_16_BIT 1
-#define CONFIG_SMC911X_BASE CS5_BASE_ADDR
-
 #define CONFIG_HAS_ETH1
 #define CONFIG_ETHPRIME
 
diff --git a/include/configs/mx53ard.h b/include/configs/mx53ard.h
index d1d5ae523f..985109e5c9 100644
--- a/include/configs/mx53ard.h
+++ b/include/configs/mx53ard.h
@@ -60,11 +60,6 @@
 
 #define CONFIG_ETHPRIME		"smc911x"
 
-/*Support LAN9217*/
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_16_BIT
-#define CONFIG_SMC911X_BASE CS1_BASE_ADDR
-
 #define CONFIG_LOADADDR		0x72000000	/* loadaddr env var */
 #define CONFIG_SYS_TEXT_BASE    0x77800000
 
diff --git a/include/configs/omap3_evm.h b/include/configs/omap3_evm.h
index 4adc66e2ab..df50325ab6 100644
--- a/include/configs/omap3_evm.h
+++ b/include/configs/omap3_evm.h
@@ -83,13 +83,6 @@
 /* USB EHCI */
 #define CONFIG_SYS_USB_FAT_BOOT_PARTITION  1
 
-/* SMSC911x Ethernet */
-#if defined(CONFIG_CMD_NET)
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_32_BIT
-#define CONFIG_SMC911X_BASE             0x2C000000
-#endif /* CONFIG_CMD_NET */
-
 /* Environment */
 #define CONFIG_PREBOOT                  "usb start"
 
diff --git a/include/configs/omap3_igep00x0.h b/include/configs/omap3_igep00x0.h
index 123de88711..1a90cb5311 100644
--- a/include/configs/omap3_igep00x0.h
+++ b/include/configs/omap3_igep00x0.h
@@ -101,15 +101,6 @@
 
 #endif
 
-/*
- * SMSC911x Ethernet
- */
-#if defined(CONFIG_CMD_NET)
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_32_BIT
-#define CONFIG_SMC911X_BASE		0x2C000000
-#endif /* (CONFIG_CMD_NET) */
-
 #define CONFIG_MTD_PARTITIONS
 #define CONFIG_SYS_MTDPARTS_RUNTIME
 
diff --git a/include/configs/omap3_logic.h b/include/configs/omap3_logic.h
index 5f8f9d7854..a5a2af19e5 100644
--- a/include/configs/omap3_logic.h
+++ b/include/configs/omap3_logic.h
@@ -236,13 +236,6 @@
 #define CONFIG_ENV_OFFSET		0x260000
 #define CONFIG_ENV_ADDR			0x260000
 
-/* SMSC922x Ethernet */
-#if defined(CONFIG_CMD_NET)
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_32_BIT
-#define CONFIG_SMC911X_BASE	0x08000000
-#endif /* (CONFIG_CMD_NET) */
-
 /* Defines for SPL */
 
 /* NAND: SPL falcon mode configs */
diff --git a/include/configs/omap3_overo.h b/include/configs/omap3_overo.h
index 8257afbe61..756b50c664 100644
--- a/include/configs/omap3_overo.h
+++ b/include/configs/omap3_overo.h
@@ -179,13 +179,6 @@
 #define CONFIG_ENV_OFFSET		0x240000
 #define CONFIG_ENV_ADDR			0x240000
 
-/* Configure SMSC9211 ethernet */
-#if defined(CONFIG_CMD_NET)
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_32_BIT
-#define CONFIG_SMC911X_BASE		0x2C000000
-#endif /* (CONFIG_CMD_NET) */
-
 /* Initial RAM setup */
 #define CONFIG_SYS_INIT_RAM_ADDR	0x4020f800
 #define CONFIG_SYS_INIT_RAM_SIZE	0x800
diff --git a/include/configs/omap3_zoom1.h b/include/configs/omap3_zoom1.h
index ed40a4b9e3..70613b5e75 100644
--- a/include/configs/omap3_zoom1.h
+++ b/include/configs/omap3_zoom1.h
@@ -153,12 +153,4 @@
 #define CONFIG_ENV_OFFSET		0x260000
 #define CONFIG_ENV_ADDR			0x260000
 
-#ifdef CONFIG_CMD_NET
-/* Ethernet (LAN9211 from SMSC9118 family) */
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_32_BIT
-#define CONFIG_SMC911X_BASE		DEBUG_BASE
-
-#endif
-
 #endif				/* __CONFIG_H */
diff --git a/include/configs/r0p7734.h b/include/configs/r0p7734.h
index ea9aa8d196..a31c6e20df 100644
--- a/include/configs/r0p7734.h
+++ b/include/configs/r0p7734.h
@@ -28,11 +28,6 @@
 #define CONFIG_BITBANGMII_MULTI
 #define CONFIG_SH_ETHER_SH7734_MII (0x00) /* MII */
 #define CONFIG_SH_ETHER_PHY_MODE PHY_INTERFACE_MODE_MII
-#ifndef CONFIG_SH_ETHER
-# define CONFIG_SMC911X
-# define CONFIG_SMC911X_16_BIT
-# define CONFIG_SMC911X_BASE (0x84000000)
-#endif
 
 /* undef to save memory	*/
 #define CONFIG_SYS_LONGHELP
diff --git a/include/configs/rsk7203.h b/include/configs/rsk7203.h
index e52756ae2b..215767cf0a 100644
--- a/include/configs/rsk7203.h
+++ b/include/configs/rsk7203.h
@@ -67,9 +67,4 @@
 #define CMT_CLK_DIVIDER	32	/* 8 (default), 32, 128 or 512 */
 #define CONFIG_SH_CMT_CLK_FREQ (CONFIG_SYS_CLK_FREQ / CMT_CLK_DIVIDER)
 
-/* Network interface */
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_16_BIT
-#define CONFIG_SMC911X_BASE (0x24000000)
-
 #endif	/* __RSK7203_H */
diff --git a/include/configs/rsk7264.h b/include/configs/rsk7264.h
index d717562f8e..11b8e0a179 100644
--- a/include/configs/rsk7264.h
+++ b/include/configs/rsk7264.h
@@ -56,9 +56,4 @@
 #define CMT_CLK_DIVIDER		32	/* 8 (default), 32, 128 or 512 */
 #define CONFIG_SH_CMT_CLK_FREQ (CONFIG_SYS_CLK_FREQ / CMT_CLK_DIVIDER)
 
-/* Network interface */
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_16_BIT
-#define CONFIG_SMC911X_BASE	0x28000000
-
 #endif	/* __RSK7264_H */
diff --git a/include/configs/rsk7269.h b/include/configs/rsk7269.h
index ac53f198c6..709563d7d8 100644
--- a/include/configs/rsk7269.h
+++ b/include/configs/rsk7269.h
@@ -55,9 +55,4 @@
 #define CMT_CLK_DIVIDER		32	/* 8 (default), 32, 128 or 512 */
 #define CONFIG_SH_CMT_CLK_FREQ (CONFIG_SYS_CLK_FREQ / CMT_CLK_DIVIDER)
 
-/* Network interface */
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_16_BIT
-#define CONFIG_SMC911X_BASE	0x24000000
-
 #endif	/* __RSK7269_H */
diff --git a/include/configs/smdkc100.h b/include/configs/smdkc100.h
index 3398b121bf..ed097b5cb6 100644
--- a/include/configs/smdkc100.h
+++ b/include/configs/smdkc100.h
@@ -174,9 +174,6 @@
  * Ethernet Contoller driver
  */
 #ifdef CONFIG_CMD_NET
-#define CONFIG_SMC911X         1       /* we have a SMC9115 on-board   */
-#define CONFIG_SMC911X_16_BIT  1       /* SMC911X_16_BIT Mode          */
-#define CONFIG_SMC911X_BASE    0x98800300      /* SMC911X Drive Base   */
 #define CONFIG_ENV_SROM_BANK   3       /* Select SROM Bank-3 for Ethernet*/
 #endif /* CONFIG_CMD_NET */
 
diff --git a/include/configs/smdkv310.h b/include/configs/smdkv310.h
index fb77db33f1..2407114086 100644
--- a/include/configs/smdkv310.h
+++ b/include/configs/smdkv310.h
@@ -88,9 +88,6 @@
 
 /* Ethernet Controllor Driver */
 #ifdef CONFIG_CMD_NET
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_BASE		0x5000000
-#define CONFIG_SMC911X_16_BIT
 #define CONFIG_ENV_SROM_BANK		1
 #endif /*CONFIG_CMD_NET*/
 
diff --git a/include/configs/twister.h b/include/configs/twister.h
index bec402ea9c..5626eb1b95 100644
--- a/include/configs/twister.h
+++ b/include/configs/twister.h
@@ -24,14 +24,6 @@
 
 #define CONFIG_HOSTNAME twister
 
-/*
- * Miscellaneous configurable options
- */
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_16_BIT
-#define CONFIG_SMC911X_BASE		0x2C000000
-#define CONFIG_SMC911X_NO_EEPROM
-
 #define	CONFIG_EXTRA_ENV_SETTINGS	CONFIG_TAM3517_SETTINGS \
 	"bootcmd=run nandboot\0"
 
diff --git a/include/configs/uniphier.h b/include/configs/uniphier.h
index 1d3bf987c0..515024c98e 100644
--- a/include/configs/uniphier.h
+++ b/include/configs/uniphier.h
@@ -28,14 +28,6 @@
 /* FLASH related */
 #define CONFIG_MTD_DEVICE
 
-#define CONFIG_SMC911X_32_BIT
-/* dummy: referenced by examples/standalone/smc911x_eeprom.c */
-#define CONFIG_SMC911X_BASE	0
-
-#ifdef CONFIG_MICRO_SUPPORT_CARD
-#define CONFIG_SMC911X
-#endif
-
 #define CONFIG_FLASH_CFI_DRIVER
 #define CONFIG_SYS_FLASH_CFI
 
diff --git a/include/configs/vct.h b/include/configs/vct.h
index d223bcebf1..9515f93da7 100644
--- a/include/configs/vct.h
+++ b/include/configs/vct.h
@@ -65,33 +65,12 @@
 #define CONFIG_SYS_LOAD_ADDR		0x80400000	/* default load address */
 
 #if defined(CONFIG_VCT_PREMIUM) || defined(CONFIG_VCT_PLATINUM)
-/*
- * SMSC91C11x Network Card
- */
-#define CONFIG_SMC911X
-#define CONFIG_SMC911X_BASE	0x00000000
-#define CONFIG_SMC911X_32_BIT
 #define CONFIG_NET_RETRY_COUNT		20
 #endif
 
 /*
  * Commands
  */
-
-/*
- * Only Premium/Platinum have ethernet support right now
- */
-#if (defined(CONFIG_VCT_PREMIUM) || defined(CONFIG_VCT_PLATINUM)) && \
-	!defined(CONFIG_VCT_SMALL_IMAGE)
-#endif
-
-/*
- * Only Premium/Platinum have USB-EHCI support right now
- */
-#if (defined(CONFIG_VCT_PREMIUM) || defined(CONFIG_VCT_PLATINUM)) && \
-	!defined(CONFIG_VCT_SMALL_IMAGE)
-#endif
-
 #if defined(CONFIG_CMD_USB)
 #define CONFIG_SUPPORT_VFAT
 
@@ -236,7 +215,6 @@ int vct_gpio_get(int pin);
  * (NOR/OneNAND) usage and Linux kernel booting.
  */
 #if defined(CONFIG_VCT_SMALL_IMAGE)
-#undef CONFIG_SMC911X
 #undef CONFIG_SYS_I2C_SOFT
 #undef CONFIG_SOURCE
 #undef CONFIG_SYS_LONGHELP
diff --git a/include/configs/vexpress_aemv8a.h b/include/configs/vexpress_aemv8a.h
index 316198a402..6203e14135 100644
--- a/include/configs/vexpress_aemv8a.h
+++ b/include/configs/vexpress_aemv8a.h
@@ -102,13 +102,7 @@
 /* Size of malloc() pool */
 #define CONFIG_SYS_MALLOC_LEN		(CONFIG_ENV_SIZE + (8 << 20))
 
-/* Ethernet Configuration */
-#ifdef CONFIG_TARGET_VEXPRESS64_JUNO
-/* The real hardware Versatile express uses SMSC9118 */
-#define CONFIG_SMC911X			1
-#define CONFIG_SMC911X_32_BIT		1
-#define CONFIG_SMC911X_BASE		(0x018000000)
-#else
+#ifndef CONFIG_TARGET_VEXPRESS64_JUNO
 /* The Vexpress64 simulators use SMSC91C111 */
 #define CONFIG_SMC91111			1
 #define CONFIG_SMC91111_BASE		(0x01A000000)
diff --git a/include/configs/vexpress_common.h b/include/configs/vexpress_common.h
index bcaf5c9426..ade7ba4a8a 100644
--- a/include/configs/vexpress_common.h
+++ b/include/configs/vexpress_common.h
@@ -55,7 +55,6 @@
 #define V2M_NOR1		(V2M_PA_CS1)
 #define V2M_SRAM		(V2M_PA_CS2)
 #define V2M_VIDEO_SRAM		(V2M_PA_CS3 + 0x00000000)
-#define V2M_LAN9118		(V2M_PA_CS3 + 0x02000000)
 #define V2M_ISP1761		(V2M_PA_CS3 + 0x03000000)
 
 /* Common peripherals relative to CS7. */
@@ -133,11 +132,6 @@
 #define CONFIG_SYS_TIMER_COUNTER	(V2M_TIMER01 + 0x4)
 #define CONFIG_SYS_TIMER_COUNTS_DOWN
 
-/* SMSC9115 Ethernet from SMSC9118 family */
-#define CONFIG_SMC911X			1
-#define CONFIG_SMC911X_32_BIT		1
-#define CONFIG_SMC911X_BASE		V2M_LAN9118
-
 /* PL011 Serial Configuration */
 #define CONFIG_PL011_SERIAL
 #define CONFIG_PL011_CLOCK		24000000
diff --git a/scripts/config_whitelist.txt b/scripts/config_whitelist.txt
index 65cee0fb70..5ee1601f0e 100644
--- a/scripts/config_whitelist.txt
+++ b/scripts/config_whitelist.txt
@@ -2044,11 +2044,6 @@ CONFIG_SLTTMR
 CONFIG_SMC91111
 CONFIG_SMC91111_BASE
 CONFIG_SMC91111_EXT_PHY
-CONFIG_SMC911X
-CONFIG_SMC911X_16_BIT
-CONFIG_SMC911X_32_BIT
-CONFIG_SMC911X_BASE
-CONFIG_SMC911X_NO_EEPROM
 CONFIG_SMC_AUTONEG_TIMEOUT
 CONFIG_SMC_USE_32_BIT
 CONFIG_SMC_USE_IOFUNCS
-- 
2.13.6

